c/tests/test_tables.c

```diff
@@ -5091,6 +5093,14 @@ test_table_collection_union(void)
         &tables, &tables_empty, node_mapping, TSK_UNION_NO_CHECK_SHARED);
     CU_ASSERT_EQUAL_FATAL(ret, 0);
 
+    // does not error on empty tables but that differ on top level metadata
+    ret = tsk_table_collection_set_metadata(
+        &tables, example_metadata, example_metadata_length);
+    CU_ASSERT_EQUAL(ret, 0);
+    ret = tsk_table_collection_union(
+        &tables, &tables_empty, node_mapping, TSK_UNION_NO_CHECK_SHARED);
```

Comment:
Maybe I've misunderstood here - but in passing `TSK_UNION_NO_CHECK_SHARED` aren't you skipping the check that you've modified?

Reply:
heh, oops, I'll fix that.